compose/foundation/foundation/src/skikoMain/kotlin/androidx/compose/foundation/Draggable.skiko.kt

```diff
@@ -0,0 +1,207 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.foundation
+
+import androidx.compose.foundation.gestures.awaitPointerSlopOrCancellation
+import androidx.compose.foundation.gestures.drag
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.composed
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.input.pointer.AwaitPointerEventScope
+import androidx.compose.ui.input.pointer.PointerButton
+import androidx.compose.ui.input.pointer.PointerEvent
+import androidx.compose.ui.input.pointer.PointerEventType
+import androidx.compose.ui.input.pointer.PointerInputChange
+import androidx.compose.ui.input.pointer.PointerKeyboardModifiers
+import androidx.compose.ui.input.pointer.PointerType
+import androidx.compose.ui.input.pointer.changedToUpIgnoreConsumed
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.input.pointer.positionChange
+import androidx.compose.ui.platform.LocalKeyboardModifiers
+import androidx.compose.ui.util.fastAll
+import kotlin.jvm.JvmInline
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.currentCoroutineContext
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.isActive
+
+@ExperimentalFoundationApi
+data class DragChange(
+    val offset: Offset = Offset.Zero,
+    val previousKeyboardModifiers: PointerKeyboardModifiers,
+    val currentKeyboardModifiers: PointerKeyboardModifiers
+) {
+    fun keyboardModifierChanged(): Boolean =
+        previousKeyboardModifiers != currentKeyboardModifiers
+}
+
+
+@JvmInline
+value class DragStartResult(val dragAmount: Offset = Offset.Zero)
+
+@OptIn(ExperimentalFoundationApi::class)
+typealias PointerFilter = PointerFilterScope.() -> Boolean
+
+@Suppress("MemberVisibilityCanBePrivate")
+object AwaitDragStart {
+    @OptIn(ExperimentalFoundationApi::class)
+    val Default: suspend AwaitPointerEventScope.(PointerEvent, PointerFilter) -> DragStartResult? =
+        { initialDown, filter ->
+            val awaitVariant = when (initialDown.changes[0].type) {
+                PointerType.Mouse -> OnSlop
+                else -> OnLongPress
```

Comment:
I have realised that even for touch we should use `slop` by default. The current `draggable` function uses `slop` to detect a drag (for slider, scroll, etc). As far as I remember, long press is used only if the component is inside a scrollable container, and we need to distinguish a parent scroll and a child drag.

Reply:
done,
we won't provide different variants to start drag for now. The hardcoded implementation is Slop.